src/lib/timeseries.spec.js

```diff
@@ -1581,3 +1582,38 @@ describe('getTitle', () => {
     })
   })
 })
+
+describe('fixSectionValues', () => {
+  it('should not do anything when section is ok', () => {
+    const section = {
+      mode: 'WALKING',
+      timestamps: [1705595674100, 1705595701100],
+      distances: [0, 20, 413],
+      distance: 433,
+      duration: 27,
+      startDate: '2024-01-18T16:31:35.100Z',
+      endDate: '2024-01-18T16:35:01.100Z',
+      speeds: [0, 0.7, 2.3],
+      avgSpeed: 1
+    }
+    expect(fixSectionsIntegrity([section])).toEqual([section])
+  })
+
+  it('should fix negative values', () => {
+    const section = {
+      mode: 'WALKING',
+      timestamps: [1705595674100, 1705595701100],
+      distances: [0, 20, 413],
+      distance: -433,
+      duration: -152,
+      startDate: '2024-01-18T16:31:35.100Z',
+      endDate: '2024-01-18T16:35:01.100Z',
+      speeds: [0, 0.7452691151217908, -2.3127397240139844],
+      avgSpeed: -1.8809647306706325
+    }
+    const fixSection = fixSectionsIntegrity([section])[0]
+    expect(fixSection.duration).toEqual(27)
```

Comment:
C'est vraiment les timestamp qui font foi? Parce qu'avec le startDate et endDate, le résultat ne ferait pas 27 mais 34.

Reply:
openpath tends to rely on timestamp, I think the ISO formatted dates are notably there for human readability. 
For the test, I changed the actual values, hence the difference between timestamps and start/end date.